angular_components/lib/material_datepicker/date_input.dart

```diff
@@ -200,22 +200,25 @@ class DateInputDirective implements OnDestroy {
     return date;
   }
 
+  Date _parseDateUsingFormat(String input, DateFormat format) {
+    try {
+      final date = Date.parseLoose(input, format);
+      return _clampDate(date);
+    } on FormatException {
+      return null;
+    } on ArgumentError {
+      // This could be a valid date format, but a date greater than
+      // DateTime's max allowed time
+      return null;
+    }
+  }
+
   /// Parse the given string as a date using one of the listed formats.
   /// Returns the parsed date, or null if the string didn't match any format.
   Date _parseDateUsingFormatList(String input, List<DateFormat> formatList) {
-    Date date;
     formatList.any((format) {
-      try {
-        date = Date.parseLoose(input, format);
-        date = _clampDate(date);
-        return true;
-      } on FormatException {
-        return false;
-      } on ArgumentError {
-        // This could be a valid date format, but a date greater than
-        // DateTime's max allowed time
-        return false;
-      }
+      date = _parseDateUsingFormat(input, format);
```

Comment:
Careful - this is the member variable `date`, not a local. We should define a local variable to avoid interfering with the state of the component.

Ideally this method would be static.

Reply:
Yes I just had an issue about this, this should be fix now but I will do some renaming work to avoid it and make it static